bodhi/server/consumers/pungi_wrapper.py

```diff
@@ -0,0 +1,348 @@
+
+        osbs_phase.dump_metadata()
+
+        # TEST phase
+        test_phase.start()
+        test_phase.stop()
+
+        self.compose.write_status("FINISHED")
+
+        self.create_latest_repo_links()
+
+        self.compose.log_info("Compose finished: %s" % self.compose.topdir)
+
+    def load_variants_config(self):
+        """
+        This is a workaround so we dont have to provide a path of the variants
+        config to pungi. We provide a generated file object which then is injected
+        into pungi compose object.
+        """
+        variants_file_obj = cStringIO.StringIO(self.variants_conf.xml)
+        parser = VariantsXmlParser(variants_file_obj)
+        self.compose.variants = parser.parse()
+        self.compose.all_variants = {}
+        for variant in self.compose.get_variants():
+            self.compose.all_variants[variant.uid] = variant
+        # After the variants object is injected into the pungi compose object
+        # we will write it to disk in the repo.
+        variants_file = self.compose.paths.work.variants_file(arch="global")
+        variants_file_obj.seek(0)
+        # Create create the variants file on disk
+        with open(variants_file, 'w') as fd:
+            shutil.copyfileobj(variants_file_obj, fd)
+            variants_file_obj.close()
+
+    def write_repo_metadata(self):
+        # write treeinfo before ISOs are created
+        for variant in self.compose.get_variants():
+            for arch in variant.arches + ["src"]:
+                pungi.metadata.write_tree_info(self.compose, arch, variant)
+
+        # write .discinfo and media.repo before ISOs are created
+        for variant in self.compose.get_variants():
+            for arch in variant.arches + ["src"]:
+                timestamp = pungi.metadata.write_discinfo(self.compose, arch, variant)
+                pungi.metadata.write_media_repo(self.compose, arch, variant, timestamp)
+
+    def create_latest_repo_links(self):
+        latest_link = True
+
+        if latest_link:
+            self.compose_dir = os.path.basename(self.compose.topdir)
+            symlink_name = "latest-%s-%s" % (
+                self.compose.conf["release_short"],
+                self.compose.conf["release_version"]
+            )
+            if self.compose.conf["release_is_layered"]:
+                symlink_name += "-%s-%s" % (
+                    self.compose.conf["base_product_short"],
+                    self.compose.conf["base_product_version"]
+                )
+            symlink = os.path.join(self.compose.topdir, "..", symlink_name)
+
+            try:
+                os.unlink(symlink)
+            except OSError as ex:
+                if ex.errno != 2:
+                    raise Exception(ex)
+            try:
+                os.symlink(self.compose_dir, symlink)
+            except Exception as ex:
+                self.compose.log_error("Couldn't create latest symlink: %s" % ex)
```

Comment:
How about `exc_info=True` to the log statement?

Reply:
We agreed during a meeting this week not to worry about smaller problems in the ```PungiWrapper``` since it comes directly from ```pungi-koji```, but I think this one could still be important to fix. I'm going to send a PR that adds a ```raise```. I'm not sure what ```self.compose.log_error()``` is, so I'm not sure that it has an ```exc_info``` arguement, but I know for sure that raise will work ☺

Reply:
https://github.com/mcurlej/bodhi/pull/24

Reply:
I sent this patch upstream too https://pagure.io/pungi/pull-request/758